src/sentry/quotas/redis.py

```diff
@@ -95,16 +101,21 @@ def get_next_period_start(interval):
         client = self.cluster.get_local_client_for_key(six.text_type(project.organization.pk))
         rejections = is_rate_limited(client, keys, args)
         if any(rejections):
+            enforce = False
             worst_case = (0, None)
             for quota, rejected in zip(quotas, rejections):
                 if not rejected:
                     continue
+                if quota.enforce:
+                    enforce = True
+                else:
+                    continue
                 delay = get_next_period_start(quota.window) - timestamp
```

Comment:
In my opinion it'd make more sense to just move [L113-115](https://github.com/getsentry/sentry/blob/908150b9dde8f20412168482a5699c5e588be79f/src/sentry/quotas/redis.py#L113-L115) into the `if` block to obviate the need for the `continue` statement. (Up to you if you'd like to make this change or not — I'm not going to block on it.)